components/mpas-framework/src/framework/mpas_field_types.inc

```diff
@@ -18,6 +18,11 @@
       ! Information used by the I/O layer
       character (len=StrKIND) :: fieldName
       character (len=StrKIND), dimension(:), pointer :: constituentNames => null()
+      ! outputFieldName and outputConstituentNames are alternative names for the field or its constituents for use in
+      ! output files.  These should only be given names different from fieldName or constituentNames for pure output
+      ! variables such as diagnostic output and never for variables present in initial conditions or restart files.
```

Comment:
@matthewhoffman, are you good with this?

Reply:
Thanks, @xylar .  This is helpful.  I would prefer to also see more info from the discussion above, specifically a reference to the specific use case this was implemented for (I think a specific example helps communicate the functionality), as well as the warning about possible collisions if multiple variables had the same `name_in_output` in the same output stream.  I agree this situation seems too difficult to try to protect against misuse, but I think it's important to document clearly what constitutes misuse.  

One other clarification for me - this new attribute is accessible through Registry, right?  But in this PR you are accessing it programmatically, right?  I was confused by that initially.

Reply:
I think a 3 line comment here is already pushing the standard practice. I hesitate to put a lengthy discussion in a file folks are unlikely to look at in detail. I don't want to set the president that this is the "right" place to document things in detail just because we lack proper documentation. If we expand this practice, this file will get out of hand.

Reply:
This PR doesn't directly make use of the new functionality either in code or in the register. It just makes sure neither breaks anything. Future, component specific PRs will actually use the functionality.

Reply:
If helpful, I might be able to find a little time later this week to have a video chat about this.

Reply:
@xylar , you're right that we don't have a good mechanism for documenting this sort of information.  I've always thought framework should have been better documented, and I'm not opposed to breaking standard practice going forward.  But I'm also ok with this PR serving as the documentation if you prefer to leave things as you have them.

Reply:
Thanks @matthewhoffman.  I fully agree that better framework documentation is needed in general and that this PR isn't a great, easy to find alternative to proper documentation.  Maybe we need a future E3SM hackathon on documenting the framework?

I looked over the comment here and your request again.  I still feel like a longer description here would just get lost.  Until there is a better place for it, I think this PR will have to suffice for documenting the purpose of this new functionality.  Hopefully, the follow-up PRs to add `Time` and `Time_bnds` coordinates will help to demonstrate the capability by example more clearly.